### PR TITLE
Fix for authorization policies and broken marketplace dashboard url

### LINKED ISSUE #352: Marketplace 'my dashboard' integration broken
**Describe the bug**
Marketplace was updated, and now they renamed the button to 'MANAGE ON PROVIDER' and it now performs a POST to the dashboard url with form body containing a JWT, similar to the activation request.

### PR DESCRIPTION
**Pull request by mservidio**
Fixes #352 

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [X] Does your submission pass tests?
2. [X] Have you lint your code locally prior to submission?

### CODE REVIEW

File: api/v1alpha/src/procurements/index.js [hunk 1]
```diff
@@ -145,4 +148,23 @@ procurements.post('/projects/:projectId/procurements/approve', async (req, res)
     });
 });
 
+procurements.post('/projects/:projectId/procurements/myProducts', async (req, res) => {
```
Comment (phriscage) [inline api/v1alpha/src/procurements/index.js:151]: Should this be a custom HTTP POST resource for the redirect e.g. `procurements:products` or `procurements:myproducts`, similar to the custom method for [`accounts:register`](https://github.com/GoogleCloudPlatform/datashare-toolkit/blob/60dd081e0ee0af19dc460bcc26cd44cdf1ca01dc/api/v1alpha/src/accounts/index.js#L638) ?

Comment (phriscage) [inline api/v1alpha/src/procurements/index.js:151]: Does this endpoint require OAS API documentation too?

Comment (mservidio) [reply api/v1alpha/src/procurements/index.js]: good point

File: api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml [hunk 2]
```diff
@@ -9,34 +9,46 @@ spec:
...
-      values:
-      - accounts.google.com
-      - https://accounts.google.com
-      - https://securetoken.google.com/$PROJECT_ID
+        methods:
+        - GET
+        paths:
+        - /v1alpha/projects/$PROJECT_ID/datasets?*
+        - /v1alpha/projects/$PROJECT_ID/procurements
+        - /v1alpha/projects/$PROJECT_ID/procurements?*
```
Comment (phriscage) [inline api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml:17]: I don't think regex (question mark '?') is an option for url paths [here](https://istio.io/v1.4/docs/reference/config/security/authorization-policy/) only wildcards '*'

File: api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml [hunk 3]
```diff
@@ -9,34 +9,46 @@ spec:
...
+        - /v1alpha/projects/$PROJECT_ID/policies/*
+    - operation:
+        methods:
+        - GET
+        - POST
+        - PUT
+        - DELETE
+        paths:
+        - /v1alpha/projects/$PROJECT_ID/datasets/*
+    when:
```
Comment (phriscage) [inline api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml:52]: Does it make more sense to group resources together vs HTTP methods? I.E. GET /datasets; GET, POST, DELETE /datasets/*, etc. I think trying to visually identify where a rule operation is applied will be harder to debug the via GET /datasets, /products; GET, POST, DELETE /datasets/*, /products/* etc. e.g. This first [operation](https://github.com/GoogleCloudPlatform/datashare-toolkit/pull/353/files#diff-6c7814bd304dd17022946e5ae309ac0b65a55bf5c3ca78ed673c5e20d1bc4ae5R15) `GET /datasets*` is also included in the last operation at the bottom of the file [here](https://github.com/GoogleCloudPlatform/datashare-toolkit/pull/353/files#diff-6c7814bd304dd17022946e5ae309ac0b65a55bf5c3ca78ed673c5e20d1bc4ae5R51) `GET /datasets/*` WDYT?

File: api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml (hunk 2 above)
Comment (mservidio) [reply api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml:17]: It's used as a literal. That call is only made with a request variable.

File: api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml (hunk 3 above)
Comment (mservidio) [reply api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml:52]: I'm not sure what's the easiest, I started by grouping resources, and then I narrowed it down to provide only the bare minimum of what's required. The GETS that you're referring to have slightly different paths.

File: api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml (hunk 2 above)
Comment (phriscage) [reply api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml:17]: ack

**Review by phriscage — APPROVED** [review]
lgtm

**Pull request merged**